Features3/tests/stepDefinitions/loginContext.js

```diff
@@ -1,34 +1,37 @@
 const { Given, When, Then } = require("@cucumber/cucumber");
 const { expect } = require("@playwright/test");
 
-Given("the user has browsed to login page", async function () {
+const emailSelector = "//input[@name='email']";
+const passwordSelector = "//input[@name='password']";
+const loginBtnSelector = "//input[@value='Login']";
+
+Given("the user has navigated to login page", async function () {
   console.log("Navigated to Login page");
   await page.goto("http://localhost:3000/");
   // await page.pause();
 });
 
-When(
-  "user enters correct credentials with email {string} and password {string}",
-  async function (email, password) {
-    const emailSelector = "//input[@name='email']";
-    const passwordSelector = "//input[@name='password']";
-    const loginBtnSelector = "//input[@value='Login']";
+When("user login with following credentials", async function (dataTable) {
+  const inputData = dataTable.hashes();
+
+  await page.fill(emailSelector, inputData[0].email);
+  await page.fill(passwordSelector, inputData[0].password);
 
-    await page.fill(emailSelector, email);
-    await page.fill(passwordSelector, password);
-    await page.click(loginBtnSelector);
-    // await page.pause();
-  }
-);
+  await page.click(loginBtnSelector);
+});
 
+// For valid credentials
 Then("user should be navigated to Contact Fox page", async function () {
   const logoutBtn = "//span[@class='hide-sm']";
-  // method 1
   const locator = page.locator(logoutBtn);
   await expect(locator).toBeVisible();
 
-  // method 2
-  // await expect(page.locator(logoutBtn)).toBeVisible();
-
   console.log("Navigated to Contact Fox page");
 });
+
+// For invalid credential
+Then("error message {string} should be shown", async function (alertMessage) {
+  const alertMsgSelector = "//div[@class='container']/div[1]";
```

Comment:
See if we can use a proper selector. Probably something that does not include `div[1]`